OrbitQt/GgpClient.cpp

```diff
@@ -0,0 +1,95 @@
+#include "GgpClient.h"
+
+#include <QDebug>
+#include <QEventLoop>
+#include <QPointer>
+#include <QProcess>
+#include <QTimer>
+
+#include "GgpInstance.h"
+#include "OrbitBase/Logging.h"
+
+const int InstanceRequestTimeoutInMilliseconds = 10'000;
+
+GgpClient::ResultOrQString<GgpClient> GgpClient::Instantiate() {
+  QProcess ggp_process{};
+  ggp_process.setProgram("ggp");
+  ggp_process.setArguments({"version"});
+  ggp_process.start(QIODevice::ReadOnly);
+  ggp_process.waitForFinished();
+
+  if (ggp_process.exitStatus() != QProcess::NormalExit ||
+      ggp_process.exitCode() != 0) {
+    return QString{
+        "Ggp command line process failed with error: %1 (exit code: %2)"}
+        .arg(ggp_process.errorString().arg(ggp_process.exitCode()));
+  }
+
+  const QByteArray process_result = ggp_process.readAllStandardOutput();
+  const QList<QByteArray> tokens = process_result.split(' ');
+
+  if (tokens.size() < 2) {
+    return QString{
+        "The current version of GGP is not supported by this integration."};
+  }
+
+  GgpClient client{};
+  client.version_ = tokens.first().toStdString();
+  return client;
+}
+
+void GgpClient::AsyncGetInstances(const InstanceCallback& callback) {
```

Comment:
GetInstancesAsync is probably better - but this is also fine.

Reply:
you are right, done